grails-app/services/com/unifina/service/SessionService.groovy

```diff
@@ -11,19 +13,29 @@ class SessionService {
 
 	KeyValueStoreService keyValueStoreService
 
-	SessionToken generateToken(SecUser user) {
-		SessionToken sk = new SessionToken(TOKEN_LENGTH, user, TTL_HOURS)
-		keyValueStoreService.setWithExpiration(sk.getToken(), sk.getUser().id.toString(), TTL_HOURS * 3600)
+	SessionToken generateToken(Userish userish) {
+		SessionToken sk = new SessionToken(TOKEN_LENGTH, userish, TTL_HOURS)
+		keyValueStoreService.setWithExpiration(sk.getToken(), userish.getClassAndId(), TTL_HOURS * 3600)
```

Comment:
IMO it would actually be nicer to drop the `getClassAndId()` method from the `Userish` interface and implement the logic in `SessionService` instead. 2 reasons:

- The inverse logic (`String` to `Userish`) is here, so for symmetry `Userish` to `String` should be too.
- `SessionService` is the only user of that logic.

For readability, the conversion in both directions could be wrapped in symmetric private methods if you like:
- `String userishToString(Userish)`
- `Userish stringToUserish(String)`